pkg/yurtctl/cmd/convert/edgenode.go

```diff
@@ -0,0 +1,399 @@
+		"The list of edge nodes wanted to be convert.(e.g. -e edgenode1,edgenode2)")
+	cmd.Flags().String("yurthub-image", "openyurt/yurthub:latest",
+		"The yurthub image.")
+	cmd.Flags().String("yurtctl-servant-image", "openyurt/yurtctl-servant:latest",
+		"The yurtctl-servant image.")
+	cmd.Flags().String("pod-manifest-path", "",
+		"Path to the directory on edge node containing static pod files.")
+	cmd.Flags().String("join-token", "", "The token used by yurthub for joining the cluster.")
+
+	return cmd
+}
+
+// Complete completes all the required options
+func (c *ConvertEdgeNodeOptions) Complete(flags *pflag.FlagSet) error {
+	enStr, err := flags.GetString("edge-nodes")
+	if err != nil {
+		return err
+	}
+	if enStr != "" {
+		c.EdgeNodes = strings.Split(enStr, ",")
+	}
+
+	yurthubImage, err := flags.GetString("yurthub-image")
+	if err != nil {
+		return err
+	}
+	c.YurthubImage = yurthubImage
+
+	ycsi, err := flags.GetString("yurtctl-servant-image")
+	if err != nil {
+		return err
+	}
+	c.YurctlServantImage = ycsi
+
+	podMainfestPath, err := flags.GetString("pod-manifest-path")
+	if err != nil {
+		return err
+	}
+	if podMainfestPath == "" {
+		podMainfestPath = os.Getenv("STATIC_POD_PATH")
+	}
+	if podMainfestPath == "" {
+		podMainfestPath = enutil.StaticPodPath
+	}
+	c.PodMainfestPath = podMainfestPath
+
+	joinToken, err := flags.GetString("join-token")
+	if err != nil {
+		return err
+	}
+	c.JoinToken = joinToken
+
+	c.clientSet, err = enutil.GenClientSet(flags)
+	if err != nil {
+		return err
+	}
+
+	kubeletSvcPath := os.Getenv("KUBELET_SVC")
+	if kubeletSvcPath == "" {
+		kubeletSvcPath = enutil.KubeletSvcPath
+	}
+	c.kubeletSvcPath = kubeletSvcPath
+
+	openyurtDir := os.Getenv("OPENYURT_DIR")
+	if openyurtDir == "" {
+		openyurtDir = enutil.OpenyurtDir
+	}
+	c.openyurtDir = openyurtDir
+
+	return nil
+}
+
+// RunConvertEdgeNode converts a standard Kubernetes node to a Yurt node
+func (c *ConvertEdgeNodeOptions) RunConvertEdgeNode() (err error) {
+	// 1. check the server version
+	if err = kubeutil.ValidateServerVersion(c.clientSet); err != nil {
+		return
+	}
+	klog.V(4).Info("the server version is valid")
+
+	nodeName, err := enutil.GetNodeName()
+	if err != nil {
+		return err
+	}
+	if len(c.EdgeNodes) > 1 || len(c.EdgeNodes) == 1 && c.EdgeNodes[0] != nodeName {
+		// 2 remote edgenode convert
+		nodeLst, err := c.clientSet.CoreV1().Nodes().List(metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		// 2.1. check the EdgeNodes and its label
+		var edgeNodeNames []string
+		for _, node := range nodeLst.Items {
+			_, ok := node.Labels[projectinfo.GetEdgeWorkerLabelKey()]
+			if !ok {
+				edgeNodeNames = append(edgeNodeNames, node.GetName())
+			}
+		}
+		for _, edgeNode := range c.EdgeNodes {
+			if !strutil.IsInStringLst(edgeNodeNames, edgeNode) {
+				klog.Errorf("Cannot do the convert, the worker node: %s is not a Kubernetes node.", edgeNode)
+				return err
+			}
+		}
+
+		// 2.2. check the state of worker nodes
```

Comment:
not check worker nodes here, so how about `check the state of EdgeNodes`

Reply:
+1 for `EdgeNodes`